File: 2-Anno/Linguaggi/Java/Pratica/Simulazioni/N40/Cliente.java
package N40;

import java.util.LinkedList;
import java.util.List;

public class Cliente {
	
	private String nome;
	private List<Noleggio> noleggi;

	public Cliente(String nome) {
		this.nome = nome;
		noleggi = new LinkedList<>();
	}

	public void addNoleggi(Noleggio noleggio) {
		noleggi.add(noleggio);
	}

	public double getCostoTotale() {
		double tot = 0.0d;

		for (Noleggio n : noleggi) {
			tot += n.getCosto();
		}

		return tot;
	}

	public String toString() {
		return nome + "\t" + getCostoTotale() + "\t" + noleggi;
	}
}

File: 2-Anno/Linguaggi/Python/Pratica/Simulazioni/N40/Cliente.py
class Cliente:
    
	def __init__(self, nome) -> None:
		self._nome = nome
		self._noleggi = []

	def addNoleggio(self, noleggio) -> None:
		self._noleggi.append(noleggio)

	def getCostoTot(self) -> float:
		tot = 0.0
		
		for noleggio in self._noleggi:
			#print(str(noleggio.getCosto()))
			tot += noleggio.getCosto()

		return tot

	def __str__(self) -> str:
		return self._nome + "\t" + str(self.getCostoTot()) + "\t" + str(len(self._noleggi)) + "\t" + ' '.join([str(i) for i in self._noleggi])
		
		